### PR TITLE
Redo fe cache

### PR DESCRIPTION
**Pull request by drwells**
Most of the fix for #675: we should now reinitialize the FE objects correctly with subdivision elements inside FEDataManager.

@xxjaehoxx I don't actually have any test code for subdivision elements but we should add some to this PR. Do you have the L2 projection code done? If so, send it to me and I will modify it to use `FECache` to check that our implementation actually works.

### CODE REVIEW

File: ibtk/src/lagrangian/FEDataManager.cpp [hunk 1]
```diff
@@ -1669,25 +1657,7 @@ FEDataManager::interpWeighted(const int f_data_idx,
...
-                // long as we have called it on this FE object at least once.
-                if (used_X_quadratures.find(key) == used_X_quadratures.end())
-                {
-                    // get_phi changes the state of the object if called
-                    // before reinit
-                    X_fe.get_phi();
-                    X_fe.reinit(elem);
-                    used_X_quadratures.insert(key);
-                }
-
```
Comment (drwells) [inline ibtk/src/lagrangian/FEDataManager.cpp]: A nice thing about this patch is that I got to delete a previous round of hacks, now that I know how to implement `FECache` correctly.

File: ibtk/include/ibtk/FECache.h [hunk 2]
```diff
@@ -54,6 +54,22 @@ namespace IBTK
...
+ * libMesh::FEBase objects stored by FECache. Multiple flags can be enabled
+ * with bitwise or operations, e.g.,
+ *
+ * @code
+ * const IBTK::FEUpdateFlags update_flags = IBTK::update_phi | IBTK::update_dphi;
+ * @endcode
+ */
+enum FEUpdateFlags
+{
+    update_default = 0,
```
Comment (boyceg) [inline ibtk/include/ibtk/FECache.h]: What is computed for `default`?

Comment (drwells) [reply ibtk/include/ibtk/FECache.h]: Nothing. I should go back and label these.

**Comment by drwells** [comment]
I checked and there is no performance regression with this patch: in fact, since we get rid of the ad-hoc initialization logic, things are about 0.03% faster :tada:

**Comment by drwells** [comment]
This PR doesn't fix all of our problems with subdivision elements, but its a good step forward. Since these changes have been approved I am going to merge to avoid getting more merge conflicts as work on FEDataManager continues.

**Pull request merged**